compiler/next/lib/AST/ErrorMessage.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/AST/ErrorMessage.h"
+
+#include <cassert>
+#include <cstdarg>
+#include <cstdlib>
+
+namespace chpl {
+namespace ast {
+
+static std::string vprint_to_string(const char* format, va_list vl) {
+  int size = 128;
+  char* buf = (char*) malloc(size);
+  int got = vsnprintf(buf, size, format, vl);
+  if (got >= size) {
```

Comment:
Have you considered using `vasprintf()`? 

https://www.man7.org/linux/man-pages/man3/asprintf.3.html

Portability might be a concern though, since these are not POSIX.

Reply:
Yeah. I think in this case I'd rather have the portable code. Having to reallocate and try again is not a big deal here.